build_tools/index_generation_s3_tar.py

```diff
@@ -180,7 +187,15 @@ def generate_index_s3(s3_client, bucket_name, upload=False):
                 "index.html",
                 ExtraArgs={"ContentType": "text/html"},
             )
-            message = "index.html successfully uploaded to the bucket root."
+            
+            # Generate a presigned URL (time-limited) using boto3
+            presigned_url = s3_client.generate_presigned_url(
+                ClientMethod="get_object",
+                Params={"Bucket": bucket_name, "Key": "index.html"},
+                ExpiresIn=3600,
+            )
+
+            message = f"index.html successfully uploaded. URL: {presigned_url}"
             gha_append_step_summary(message)
```

Comment:
I'm not very familiar with the details of this "time limited presigned URL" concept, but it looks like something sensitive that should not be logged. The file is public, so shouldn't we use a public URL? We could construct that URL from just the bucket name. We have code in a few of these scripts that does that already:
* https://github.com/ROCm/TheRock/blob/dc05d637054ad197c84b00e24b6262af0ec797c6/build_tools/github_actions/post_build_upload.py#L216-L218
* https://github.com/ROCm/TheRock/blob/dc05d637054ad197c84b00e24b6262af0ec797c6/build_tools/install_rocm_from_artifacts.py#L205-L210